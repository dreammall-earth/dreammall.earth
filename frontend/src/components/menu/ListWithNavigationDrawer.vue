<template>
  <NavigationDrawer :modelValue="drawer" @update:modelValue="updateDrawer">
    <ListElement :items="items" />
  </NavigationDrawer>
</template>

<script lang="ts" setup>
import { ref, defineProps, defineEmits } from 'vue'
import NavigationDrawer from './NavigationDrawer.vue'
import ListElement from './ListElement.vue'

const props = defineProps({
  drawer: {
    type: Boolean,
    required: true
  }
})

const emits = defineEmits(['update:drawer'])

const items = ref([
  { title: 'LOLLY Krypto Entwicklung', fullWidth: false },
  { title: 'Aachener Freunde Treff', fullWidth: false },
  { title: 'Building A Wooden Guitar', fullWidth: false },
  { title: 'Co Working Space Manufaktur', fullWidth: true },
  { title: 'Intervallfasten Kalender', fullWidth: false },
  { title: 'Lomografie 2050', fullWidth: true },
  { title: 'Freies Treffen U50', fullWidth: false },
])

const updateDrawer = (value: boolean) => {
  emits('update:drawer', value)
}
</script>

<style scoped>
.list-container {
  margin-top: 20px;
}
</style>

