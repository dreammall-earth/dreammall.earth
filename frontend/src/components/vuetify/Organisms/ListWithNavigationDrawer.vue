<template>
    <NavigationDrawer :model-value="drawer" :location="location" @update:model-value="updateDrawer">
      <SearchField
        v-model="search"
        label="Open Tables, Jobs"
        prepend-inner-icon="mdi-tune"
      ></SearchField>
      <v-list>
        <div class="mx-4">{{ $t('menu.roomList') }}</div>
        <ListElement :items="filteredItems" @item-click="updateDrawer(false)" />
      </v-list>
    </NavigationDrawer>
  </template>
  
  <script lang="ts" setup>
  import { ref, computed } from 'vue'
  import { VAvatar, VIcon, VImg } from 'vuetify/components'
  import SearchField from '../search/SearchField.vue'
  import ListElement from './../Atoms/ListElement.vue'
  import NavigationDrawer from './../Atoms/NavigationDrawer.vue'
  defineProps({
    drawer: {
      type: Boolean,
      required: true,
    },
    location: {
      type: String as () => 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start' | undefined,
      required: false,
      default: 'right',
    },
  })
  const emits = defineEmits(['update:drawer'])
  const search = ref('')
  const items = [
    {
      title: 'LOLLY Krypto Entwicklung',
      subtitle: 'Krypto fÃ¼r Alle',
      prepend: VAvatar,
      prependProps: { color: 'primary', icon: 'mdi-lock' },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Aachener Freunde Treff',
      subtitle: 'Aachener Freunde Treff',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/40',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Building A Wooden Guitar',
      subtitle: 'Building A Wooden Guitar',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/39',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Co Working Space Manufaktur',
      subtitle: 'Co Working Space Manufaktur',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/41',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
    {
      title: 'Intervallfasten Kalender',
      subtitle: 'Intervallfasten Kalender',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/42',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical-circle', class: 'append-icon' },
    },
    {
      title: 'Lomografie 2050',
      subtitle: 'Lomografie 2050',
      prepend: VImg,
      prependProps: {
        src: 'https://picsum.photos/46',
        referrerpolicy: 'no-referrer',
        width: 40,
        height: 40,
      },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical-circle', class: 'append-icon' },
    },
    {
      title: 'Freies Treffen U50',
      subtitle: 'Freies Treffen U50',
      prepend: VAvatar,
      prependProps: { color: 'primary', image: 'https://picsum.photos/46' },
      append: VIcon,
      appendProps: { icon: 'mdi-dots-vertical', class: 'append-icon' },
    },
  ]
  const filteredItems = computed(() => {
    if (!search.value) {
      return items
    }
    return items.filter((item) => item.title.toLowerCase().includes(search.value.toLowerCase()))
  })
  const updateDrawer = (value: boolean) => {
    emits('update:drawer', value)
  }
  </script>