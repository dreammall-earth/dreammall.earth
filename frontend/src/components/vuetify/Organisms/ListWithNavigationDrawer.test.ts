import { shallowMount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'

import ListWithNavigationDrawer from './ListWithNavigationDrawer.vue'

describe('ListWithNavigationDrawer', () => {
  let wrapper: ReturnType<typeof shallowMount>

  beforeEach(() => {
    wrapper = shallowMount(ListWithNavigationDrawer, {
      props: { drawer: true },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('toggles drawer state', async () => {
    const drawer = wrapper.findComponent({ name: 'NavigationDrawer' })
    expect(drawer.props('modelValue')).toBe(true)
    await wrapper.setProps({ drawer: false })
    await wrapper.vm.$nextTick()
    expect(drawer.props('modelValue')).toBe(false)
  })
})