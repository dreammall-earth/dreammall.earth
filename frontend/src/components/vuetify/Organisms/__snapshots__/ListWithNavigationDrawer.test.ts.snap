// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListWithNavigationDrawer > renders correctly 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    
    <nav
      class="v-navigation-drawer v-navigation-drawer--right v-navigation-drawer--temporary v-navigation-drawer--active v-theme--light v-navigation-drawer--mobile menu-drawer-top"
      data-v-3308ee86=""
      style="right: 0px; z-index: 1004; transform: translateX(0%); position: fixed; transition: none !important; height: calc(100% - 0px - 0px); top: 0px; bottom: 0px;"
    >
      <!---->
      <!---->
      <div
        class="v-navigation-drawer__content"
      >
        
        <!--
    &lt;SearchField
      v-model="search"
      label="Open Tables, Jobs"
      prepend-inner-icon="mdi-tune"
    &gt;&lt;/SearchField&gt;
    -->
        <div
          class="v-list v-theme--light bg-transparent v-list--density-default v-list--one-line"
          data-v-3308ee86=""
          role="listbox"
          tabindex="0"
        >
          
          <div
            class="mx-4"
            data-v-3308ee86=""
          >
            $t('menu.roomList')
          </div>
          <div
            class="v-card v-theme--light v-card--density-default v-card--variant-elevated mx-auto"
            data-v-3308ee86=""
            data-v-fd6c8fbc=""
            style="width: 400px;"
          >
            <!---->
            <div
              class="v-card__loader"
            >
              <div
                aria-hidden="true"
                aria-valuemax="100"
                aria-valuemin="0"
                class="v-progress-linear v-theme--light v-locale--is-ltr"
                role="progressbar"
                style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
              >
                <!---->
                <div
                  class="v-progress-linear__background"
                  style="width: 100%;"
                />
                <transition-stub
                  appear="false"
                  css="true"
                  name="fade-transition"
                  persisted="false"
                >
                  <div
                    class="v-progress-linear__indeterminate"
                  >
                    
                    <div
                      class="v-progress-linear__indeterminate long"
                    />
                    <div
                      class="v-progress-linear__indeterminate short"
                    />
                    
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <!---->
            
            <div
              class="v-list v-theme--light bg-transparent v-list--density-default v-list--two-line"
              data-v-fd6c8fbc=""
              role="listbox"
              tabindex="0"
            >
              
              
              
              
            </div>
            
            <!---->
            
            <!---->
            <span
              class="v-card__underlay"
            />
            
          </div>
          
        </div>
        
      </div>
      <!---->
    </nav>
    <transition-stub
      appear="false"
      css="true"
      name="fade-transition"
      persisted="false"
    >
      <div
        class="v-navigation-drawer__scrim"
        data-v-3308ee86=""
        style="z-index: 1003;"
      />
    </transition-stub>
    
    
  </div>
</div>
`;
